src/main/java/org/apache/maven/plugins/dependency/GetClassesMojo.java

```diff
@@ -0,0 +1,406 @@
+
+    /**
+     * The version of the artifact to download. Ignored if {@link #artifact} is used.
+     */
+    @Parameter( property = "version" )
+    private String version;
+
+    /**
+     * The classifier of the artifact to download. Ignored if {@link #artifact} is used.
+     *
+     * @since 2.3
+     */
+    @Parameter( property = "classifier" )
+    private String classifier;
+
+    /**
+     * The packaging of the artifact to download. Ignored if {@link #artifact} is used.
+     */
+    @Parameter( property = "packaging", defaultValue = "jar" )
+    private String packaging = "jar";
+
+    /**
+     * Repositories in the format id::[layout]::url or just url, separated by comma. ie.
+     * central::default::https://repo.maven.apache.org/maven2,myrepo::::https://repo.acme.com,https://repo.acme2.com
+     */
+    @Parameter( property = "remoteRepositories" )
+    private String remoteRepositories;
+
+    /**
+     * A string of the form groupId:artifactId:version[:packaging[:classifier]].
+     */
+    @Parameter( property = "artifact" )
+    private String artifact;
+
+    @Parameter( defaultValue = "${project.remoteArtifactRepositories}", readonly = true, required = true )
+    private List<ArtifactRepository> pomRemoteRepositories;
+
+    /**
+     * Download transitively, retrieving the specified artifact and all of its dependencies.
+     */
+    @Parameter( property = "transitive", defaultValue = "true" )
+    private boolean transitive = true;
+
+    /**
+     * Skip plugin execution completely.
+     *
+     * @since 2.7
+     */
+    @Parameter( property = "mdep.skip", defaultValue = "false" )
+    private boolean skip;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException
+    {
+        ProjectBuildingRequest buildingRequest = buildBuildingRequest();
+        DefaultDependableCoordinate coordinate = getCoordinate();
+
+        try
+        {
+            getLog().info( "Retrieving classes of dependencies for " + artifact );
+            if ( isTransitive() )
+            {
+                Iterable<ArtifactResult> artifacts = getDependencyResolver()
+                        .resolveDependencies( buildingRequest, coordinate, null );
+
+                for ( ArtifactResult result : artifacts )
+                {
+                    printClassesFromArtifactResult( result );
+                }
+            }
+            else
+            {
+                ArtifactResult result = getArtifactResolver()
+                        .resolveArtifact( buildingRequest, toArtifactCoordinate( coordinate ) );
+
+                printClassesFromArtifactResult( result );
+            }
+        }
+        catch ( ArtifactResolverException | DependencyResolverException | IOException e )
+        {
+            throw new MojoExecutionException( "Couldn't download artifact: " + e.getMessage(), e );
+        }
+    }
+
+    public void printClassesFromArtifactResult( ArtifactResult result )
```

Comment:
why is this public? Methods should have the most restrictive access possible.

Reply:
fixed